pipeline/build/build.go

```diff
@@ -30,16 +33,44 @@ func (Pipe) Description() string {
 func (Pipe) Run(ctx *context.Context) error {
 	for _, build := range ctx.Config.Builds {
 		log.WithField("build", build).Debug("building")
+		if err := checkMain(ctx, build); err != nil {
+			return err
+		}
 		if err := runPipeOnBuild(ctx, build); err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
+func checkMain(ctx *context.Context, build config.Build) error {
+	var glob = build.Main
+	if !strings.HasSuffix(glob, "main.go") {
+		// TODO: in real live , glob will never be empty. Maybe this is worth
+		// guarding here anyway
+		glob = glob + "/" + "*.go"
+	}
+	log.Debugf("glob is %s", glob)
+	files, err := zglob.Glob(glob)
+	if err != nil {
+		return errors.Wrap(err, "failed to find go files")
+	}
+	log.WithField("files", files).Debug("go files")
+	for _, file := range files {
+		bts, err := ioutil.ReadFile(file)
+		if err != nil {
+			return errors.Wrapf(err, "failed to read file: %s", file)
+		}
+		if strings.Contains(string(bts), "func main() {") {
```

Comment:
What if the code is not formatted using `go fmt`? This will not work. Wouldn't it be better to use the parser somehow? For example see the first answer in https://stackoverflow.com/questions/41629293/how-do-i-list-the-public-methods-of-a-package-in-golang

Reply:
good idea.